pkg/controller/servicebindingrequest/sbrcontroller_test.go

```diff
@@ -0,0 +1,181 @@
+package servicebindingrequest
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/log"
+)
+
+func Test_newSBRPredicate(t *testing.T) {
+
+	// keep the predicate around
+	pred := newSBRPredicate(log.NewLog("test-log"))
+
+	// the expected behavior is that every create event triggers a reconciliation
+	t.Run("create", func(t *testing.T) {
+		if got := pred.Create(event.CreateEvent{}); !got {
+			t.Errorf("newSBRPredicate() = %v, want %v", got, true)
+		}
+	})
+
+	// update exercises changes that should or not trigger the reconciliation
+	t.Run("update", func(t *testing.T) {
+		sbrA := &v1alpha1.ServiceBindingRequest{
+			Spec: v1alpha1.ServiceBindingRequestSpec{
+				BackingServiceSelector: v1alpha1.BackingServiceSelector{
+					Group:       "test",
+					Version:     "v1alpha1",
+					Kind:        "TestHost",
+					ResourceRef: "",
+				},
+			},
+		}
+		sbrB := &v1alpha1.ServiceBindingRequest{
+			Spec: v1alpha1.ServiceBindingRequestSpec{
+				BackingServiceSelector: v1alpha1.BackingServiceSelector{
+					Group:       "test",
+					Version:     "v1",
+					Kind:        "TestHost",
+					ResourceRef: "",
+				},
+			},
+		}
+
+		tests := []struct {
+			name string
+			want bool
+			a    runtime.Object
+			b    runtime.Object
+		}{
+			{name: "same-spec", want: false, a: sbrA, b: sbrA},
+			{name: "changed-spec", want: true, a: sbrA, b: sbrB},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				if got := pred.Update(event.UpdateEvent{ObjectOld: tt.a, ObjectNew: tt.b}); got != tt.want {
+					t.Errorf("newSBRPredicate() = %v, want %v", got, tt.want)
+				}
+			})
+		}
+	})
+
+	// delete verifies that SBRs will be reconciled prior to its deletion
+	t.Run("delete", func(t *testing.T) {
+		tests := []struct {
+			name           string
+			want           bool
+			confirmDeleted bool
+		}{
+			// FIXME: validate whether this is the behavior we want
+			{name: "delete-not-confirmed", confirmDeleted: false, want: true},
+			{name: "delete-confirmed", confirmDeleted: true, want: false},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				if got := pred.Delete(event.DeleteEvent{DeleteStateUnknown: tt.confirmDeleted}); got != tt.want {
+					t.Errorf("newSBRPredicate() = %v, want %v", got, tt.want)
+				}
+			})
+		}
+	})
+}
+
+func Test_newGVKPredicate(t *testing.T) {
```

Comment:
Same here.

Reply:
Solved in cc0d39e35f1f219cfd3f041d482c75186f870309.